idb/postgres/postgres.go

```diff
@@ -85,6 +86,30 @@ func openPostgres(db *sql.DB, opts *idb.IndexerDbOptions, logger *log.Logger) (p
 	return
 }
 
+func txnWithRetry(f func() error) error {
+	maxDelay := time.Minute
+	delay := time.Millisecond
+
+	for {
+		err := f()
+
+		// If not serialization error.
+		var pqerr *pq.Error
+		if !errors.As(err, &pqerr) || (pqerr.Code != "40001") {
+			return err
+		}
+
+		fmt.Println("retrying transaction")
+		// Randomize delay to avoid conflicts.
+		time.Sleep(time.Duration(rand.Int63n(int64(delay))))
+
+		delay *= 2
+		if delay > maxDelay {
+			maxDelay = delay
```

Comment:
return serialize error when delay > maxDelay?

Reply:
and why delay *= 2?

Reply:
You probably right, we don't need a delay.